### PR TITLE
Refactor IsValid() in did/web/web.go

### PR DESCRIPTION
**Pull request by vlad-timofeev**
## Description

* Change return type of  `IsValid()` to `error`, and rename the function to `Validate()`.
* Add `context.Context` as it's first parameter and propagate it to the network call.
* Add a unit test that calls `Validate()` function.
* Update production code and tests code to account for the method signature changes.

## Testing

Ran `mage cbt` without any errors.

## References

Related to  #395 .

### CODE REVIEW

**Comment by michaelneale** [comment]
nice to see a test case too, cool

File: did/web/web.go [hunk 1]
```diff
@@ -29,9 +30,13 @@ const (
...
 )
 
-func (d DIDWeb) IsValid() bool {
-	_, err := d.resolveDocBytes()
-	return err == nil
+// Validate return nil if DID is valid, otherwise the validation error.
+func (d DIDWeb) Validate(ctx context.Context) error {
+	_, err := d.resolveDocBytes(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to validate DID")
```
Comment (andresuribe87) [inline did/web/web.go:37]:
```suggestion
		return errors.Wrap(err, "resolving doc bytes")
```

File: did/web/web_test.go [hunk 2]
```diff
@@ -135,3 +136,23 @@ func TestDIDWebCreateDocFileBytes(t *testing.T) {
...
+		defer gock.Off()
+
+		err := didWebToBeResolved.Validate(context.Background())
+		assert.NoError(tt, err)
+	})
+
+	t.Run("Unresolvable Path - Validate DID", func(tt *testing.T) {
+		err := didWebNotADomain.Validate(context.Background())
+		assert.Error(tt, err)
+		assert.Contains(tt, err.Error(), "failed to validate")
```
Comment (andresuribe87) [inline did/web/web_test.go:155]: Prefer using
```suggestion
		assert.ErrorContains(tt, err, "failed to validate")
```

**Review by andresuribe87 — APPROVED** [review]
Thanks so much!  A couple minor nits

**Comment by vlad-timofeev** [comment]
Looks like this PR now has conflicts with the mainline. I have just made an attempt to resolve them from the Github UI.

**Pull request merged**